compiler/src/iree/compiler/DispatchCreation/HoistEncodingOps.cpp

```diff
@@ -204,7 +204,14 @@ void HoistEncodingOpsPass::runOnOperation() {
   });
   IRRewriter rewriter(ctx);
   for (auto setEncodingOp : candidates) {
+    Value src = setEncodingOp.getSource();
+    if (getConstantIntValue(src).has_value() &&
```

Comment:
you want more than integers - you only care that the source is ConstantLike and can check for that (something like `getDefiningOp()->hasTrait<ConstantLike>()`, probably)

Reply:
Oh, thanks for pointing it out. I missed that the method only works for integers.